polynote-frontend/polynote/ui/layout/splitview.ts

```diff
@@ -85,11 +105,15 @@ export class SplitView extends Disposable {
     private leftDragger: Dragger;
     private rightDragger: Dragger;
 
-    constructor(leftPane: Pane, private center: TagElement<"div">, rightPane: Pane) {
+    private readonly leftView: StateHandler<ViewPreferences["leftPane"]>;
+    private readonly stickyLeftMenu: StateHandler<LeftBarPreferences["stickyLeftMenu"]>;
+
+    constructor(nbList: NotebookList, tableOfContents: TableOfContents, private center: TagElement<"div">, rightPane: Pane, dispatcher: ServerMessageDispatcher) {
```

Comment:
Hmmm... I think the intent is that `SplitView` is reasonably general, rather than being coupled to/responsible for the precise UI structure.

I guess, because `SplitView` handles the collapsing (and tabbing) of the left pane, it does have to be responsible for that. But, maybe it can take a general array of left panes rather than two specific ones?

I also don't think it should have to take the `ServerMessageDispatcher` here. Let's figure out if that can be avoided with design decisions.

Reply:
Good call - the dispatcher definitely should not have been there. For the `SplitView`, I totally agree but struggle to abstract it out much as it's very closely intertwined with the collapsing and tabbing (as clicking on any of the elements has to tab a specific element and access the state, etc.). I abstracted it out into an object, but wasn't able to easily make the entire component agnostic to what is inside of the left pane. Is this sufficient, or do you think more design work should be done on this?